File: spLLT/data/cmp_facto.py
import fileinput
import re

import vextractor
import latexprint as lp

spllt_facto_time = '\[>\] \[factorize\] time:'
ma87_facto_time  = 'Factor took'

blocksizes = [64, 128, 256, 512]

outputdir = 'cn255'

for mat in fileinput.input():

    print mat
    pbl = re.sub(r'/', '_', mat)
    pbl = pbl.rstrip()
    print pbl

    # spLLT
    spllt_t_facto = []
    for blocksize in blocksizes:
        print "blocksize: ",blocksize
        datafile = outputdir + '/' + 'spllt_starpu' + '/' + pbl + '_NCPU-27' + '_NB-' + str(blocksize)
        # print datafile
        f = open(datafile)
        v = vextractor.get_value(f, spllt_facto_time)
        f.close()

        spllt_t_facto.append(float(v))

    # MA87
    ma87_t_facto = []
    for blocksize in blocksizes:
        print "blocksize: ",blocksize
        datafile = outputdir + '/' + 'ma87' + '/' + pbl + '_NCPU-27' + '_NB-' + str(blocksize)
        # print datafile
        f = open(datafile)
        v = vextractor.get_value(f, ma87_facto_time)
        f.close()

        ma87_t_facto.append(float(v))
    
    print spllt_t_facto
    print ma87_t_facto
        

File: SpLLT/data/cmp_facto.pl
#!/usr/bin/perl

use ValueExtractor;

# directory where runs are kept
my $spllt_output_dir='spllt_starpu';
my $ma87_output_dir='ma87';

# my $facto_time_string = '[>] [spllt_stf_factorize] time:';
my $spllt_facto_time_string = '\[>\] \[factorize\] time:';
my $ma87_facto_time_string  = 'Factor took';

# print "$facto_time_string\n";

my $suffix = "_NCPU-27_NB-256";

printf("%30s%12s%12s\n", " ", "spLLT", "MA87");

# Loop over problems input on stdin
while(<>) {
   chomp;
   if(m/^#/) { next; }
   
   my $prob = $_;
   my $fprob = $prob;
   $fprob =~ s/\//_/g;

   # print "$_\n";
   # print "$fprob\n";
   
   $t_spllt = get_value("${spllt_output_dir}", $fprob, "_NCPU-27_NB-256", $spllt_facto_time_string, "mean");
   $t_ma87  = get_value("${ma87_output_dir}" , $fprob, "_NCPU-27_NB-256", $ma87_facto_time_string , "mean");

   printf("%30s", $prob);
   printf("%12.3e", $t_spllt);
   printf("%12.3e", $t_ma87 );

   # print "$t_spllt\n";
   # print "$t_ma87\n";
   printf("\n");
}
